internal/librarian/internal/rust/create.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rust
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/googleapis/librarian/internal/command"
+	"github.com/pelletier/go-toml/v2"
+)
+
+// RustCreator interface used for mocking in tests.
+type RustCreator interface {
```

Comment:
What's the reason that we need mocking? Could you just call sidekick.PrepareCargoWorkspace directly?